### PR TITLE
Fix/protractor implementation

### PR DESCRIPTION
**Pull request by SkxPhan**
Pull request to fix issue #1.

1. Fix resample function to ensure correct number of points
2. Fix vectorizeStroke function based on Protractor pseudo code paper
3. Fix optimalCosineDistance function
4. Adapt and add test case for the _recognize_ function

### CODE REVIEW

**Review by htfy96 — APPROVED** [review]
Sorry for the delay. Checked all proposed changes and they all look great. Thanks for your contribution!

File: test/test.cpp [hunk 1]
```diff
@@ -13,7 +13,17 @@ TEST_CASE("1 instance")
...
+}
+
+TEST_CASE("2 instances")
+{
+    vector<Stroke> strokes{ Stroke({ { 0., 0. }, { 0., 1. }, { 1., 1. } }, Orientation::Sensitive),
+                            Stroke({ { 0., 0. }, { 1., 1. }, { 1., 0. } }, Orientation::Sensitive) };
+    Stroke testStroke{ { { 0., 0. }, { 0., 0.9 }, { 0.02, 0.91 }, { 1., 1. } }, Orientation::Sensitive };
+    auto [it, score] = recognize(testStroke, strokes.begin(), strokes.end());
+    CHECK(it == strokes.begin());
+    CHECK_THAT(score, Catch::Matchers::WithinAbs(17.6808f, 0.001f));
```
Comment (htfy96) [inline test/test.cpp]: This threshold is too stringent and caused the test to fail in Github CI. Can you change it to `0.01f`?

**Review by htfy96 — CHANGES_REQUESTED** [review]

File: test/test.cpp (hunk 1 above)
Comment (SkxPhan) [reply test/test.cpp]: Done!

**Comment by htfy96** [comment]
Hi, the CIs are still failing. Would you like to a look when having time? Alternatively I can also help debugging

**Comment by SkxPhan** [comment]
Hi, I think the first test failed because of the perfect match. Therefore score reach an infinite value due to a zero division happening in the function _recognize_:
```cpp
recognize(const Stroke& target, ItT templateItBegin, ItT templateItEnd)
{
    float maxScore = 0;
    ItT matchedIt = templateItEnd;
    for (ItT templateIt = templateItBegin; templateIt != templateItEnd; ++templateIt) {
        const float d = detail::optimalCosineDistance(target.getVectorizedStroke(), templateIt->getVectorizedStroke());
        const float score = 1.0f / d;
        if (score > maxScore) {
            maxScore = score;
            matchedIt = templateIt;
        }
    }
    return { matchedIt, maxScore };
}
```
To cope with this specific issue and avoid any undefined behavior, I added an if statement to handle the case of d = 0 and modified the test case.

**Review by htfy96 — APPROVED** [review]

**Pull request merged**